package.json

```diff
@@ -81,29 +80,51 @@
     "@types/react-dom": "^16.9.5",
     "@types/semver": "^7.1.0",
     "@types/tmp": "0.1.0",
+    "@typescript-eslint/eslint-plugin": "^3.7.0",
+    "@typescript-eslint/parser": "^3.7.0",
     "chokidar": "^3.3.1",
     "coveralls": "^3.0.9",
     "electron": "8.2.4",
     "enzyme": "^3.11.0",
     "enzyme-adapter-react-16": "^1.15.2",
     "enzyme-to-json": "^3.4.4",
+    "eslint": "^7.5.0",
+    "eslint-config-prettier": "^6.11.0",
+    "eslint-plugin-prettier": "^3.1.4",
+    "eslint-plugin-react": "^7.20.3",
     "fetch-mock-jest": "^1.1.0-beta.3",
+    "husky": "^4.2.5",
     "jest": "^25.1.0",
     "less": "^3.11.1",
+    "lint-staged": "^10.2.11",
     "log-symbols": "^3.0.0",
     "node-abi": "^2.18.0",
     "node-fetch": "^2.6.0",
     "npm-run-all": "^4.1.5",
     "parcel-bundler": "^1.12.4",
+    "prettier": "^2.0.5",
     "react-test-renderer": "^16.13.0",
     "rimraf": "^3.0.0",
     "standard": "^14.3.1",
     "stylelint": "^13.2.0",
     "stylelint-config-standard": "^20.0.0",
     "ts-jest": "^25.2.1",
-    "tslint": "^6.0.0",
-    "tslint-microsoft-contrib": "^6.2.0",
-    "tslint-react": "^4.2.0",
     "typescript": "^3.8.2"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "./**/*.{js,ts,tsx}": [
+      "npm run lint:ts"
+    ],
+    "./static/show-me/**/*.js": [
+      "npm run lint:templates"
+    ],
+    "./src/less/*.less": [
+      "npm run lint:style"
+    ]
```

Comment:
This isn't going to work quite the way you want it to (I know this from experience). What should happen is that only files that are changed should get run through the appropriate linter. In practice, all of the files specified in the lint subcommand will always get processed. I'd replace the `npm run` commands with the bare linter commands (sans the file globs).